### ISSUE TITLE
Auth with JIRA access token

### ISSUE THREAD

**Issue by JuergenLandt** [Icinga/icingaweb2-module-jira#66]
As an alternative to login with user and password, login with a token should be possible.

`[api]
host = "jira.example.com"
; port = 443
; path = "/"
; scheme = "https"
; username = "icinga"
; password = "***"
token = "****************************************************"

`

**Comment by verboEse** [comment]
AFAIC, you don't need to provide a password when using the Jira-token. In our setup, it's sufficient to configure the token in the password field.

**Comment by Littlericket** [comment]
@verboEse Unfortunately, no.
![image](https://github.com/Icinga/icingaweb2-module-jira/assets/1573629/9fcb1f7d-cd76-4362-9bcc-68c2e2888506)

**Comment by verboEse** [comment]
Sorry, I wasn't clear. You have to provide the username of course. But you only have to provide either password *OR* token.

**Comment by Littlericket** [comment]
@verboEse I don't think that this will work on Server deployments, since the authorization is done via bearer-token (PAT or access token) and not basic-auth. For RestApi (https://github.com/Icinga/icingaweb2-module-jira/blob/main/library/Jira/RestApi.php#L434), there needs to be an addition to use the bearer token, like (for example)
```
        if($config_setting == "bearer-token") {
            $auth = null;
            $headers[] = "Authorization: Bearer ".$token;
        }
```

@theFeu maybe this is an interesting addition? For the config form, there can be an autosubmit for the type of authorization, username/password or via token. This will switch adding the above mentioned header and removes the basic-authentication. Instead of storing username / password, you will only store the access token.

**Comment by theFeu** [comment]
Heyhey, I'm no longer active in this project.

@raviks789 might be a better person to ask!

**Comment by raviks789** [comment]
@Littlericket I am currently occupied with other projects. And I am not sure if this will be included in the next release. But I will look into this later.
